### PR TITLE
feat: allow different bot name to auto-merge

### PR DESCRIPTION
**Pull request by viestat**
This Pull Request fulfills the following requirements:

- [x] The commit message follows our guidelines.
- [x] Tests for the changes have been added if needed.
- [x] Does not affect documentation or it has been added or updated.

Original from https://github.com/ridedott/dependabot-auto-merge-action/pull/149 by @Qcastel .

Took it here as it needed to be resolved

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,6 +59,21 @@ auto-merge:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 ```
 
+### Enable auto-merge for a different bot
+
+You may have another bot that also raises PR against your repo and you want to
```
Comment (merlinnot) [inline README.md]:
```suggestion
You may have another bot that also creates PRs against your repository and you want to
```

File: README.md [hunk 2]
```diff
@@ -59,6 +59,21 @@ auto-merge:
...
+auto-merge. By default, this github action will assume the bot is dependabot.
```
Comment (merlinnot) [inline README.md]:
```suggestion
automatically merge those. By default, this GitHub Action assumes the bot is [`dependabot`](https://dependabot.com/).
```

File: README.md [hunk 3]
```diff
@@ -59,6 +59,21 @@ auto-merge:
...
+You can override the bot name by setting up the value `GITHUB_LOGIN`:
```
Comment (merlinnot) [inline README.md]:
```suggestion
You can override the bot name by changing the value of `GITHUB_LOGIN` parameter:
```

File: README.md [hunk 4]
```diff
@@ -59,6 +59,21 @@ auto-merge:
...
+
+```yaml
+steps:
+  - name: Auto merge
+    uses: ridedott/dependabot-auto-merge-action@master
+    with:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      GITHUB_LOGIN: my-awesome-bot-r2d2
```
Comment (merlinnot) [inline README.md]: Sort :)

File: action.yml [hunk 5]
```diff
@@ -7,6 +7,11 @@ inputs:
   GITHUB_TOKEN:
     description: 'A GitHub token.'
     required: true
+  GITHUB_LOGIN:
+    description:
+      'The github login on which you want the auto-merge to be enabled'
```
Comment (merlinnot) [inline action.yml]:
```suggestion
      'The GitHub login for which automatic merges are enabled.'
```

File: src/eventHandlers/checkSuite/index.spec.ts [hunk 6]
```diff
@@ -16,7 +16,7 @@ const COMMIT_HEADLINE = 'Update test';
 
 const octokit = new GitHub('SECRET_GITHUB_TOKEN');
 
-jest.spyOn(core, 'info').mockImplementation();
+const infoSpy = jest.spyOn(core, 'info').mockImplementation();
 
 const warningSpy = jest.spyOn(core, 'warning').mockImplementation();
```
Comment (merlinnot) [inline src/eventHandlers/checkSuite/index.spec.ts]: Do we need a newline here?

File: src/eventHandlers/checkSuite/index.spec.ts [hunk 7]
```diff
@@ -114,7 +114,207 @@ describe('check Suite event handler', (): void => {
...
+      });
+
+    await checkSuiteHandle(octokit, 'dependabot-preview[bot]');
+
+    expect(infoSpy).toHaveBeenCalledWith(
+      'Pull request is not in a mergeable state: CONFLICTING.',
+    );
+  });
+
+  it('does not approve pull request that is already merged', async (): Promise<
```
Comment (merlinnot) [inline src/eventHandlers/checkSuite/index.spec.ts]:
```suggestion
  it('does not approve pull requests that are already merged', async (): Promise<
```

File: src/eventHandlers/checkSuite/index.spec.ts [hunk 8]
```diff
@@ -114,7 +114,207 @@ describe('check Suite event handler', (): void => {
...
+          },
+        },
+      });
+
+    await checkSuiteHandle(octokit, 'dependabot-preview[bot]');
+
+    expect(infoSpy).toHaveBeenCalledWith('Pull request is already merged.');
+  });
+
+  it('does not approve pull request which status is not clean', async (): Promise<
```
Comment (merlinnot) [inline src/eventHandlers/checkSuite/index.spec.ts]:
```suggestion
  it('does not approve pull requests for which status is not clean', async (): Promise<
```

File: src/eventHandlers/checkSuite/index.spec.ts [hunk 9]
```diff
@@ -114,7 +114,207 @@ describe('check Suite event handler', (): void => {
...
+      });
+
+    await checkSuiteHandle(octokit, 'dependabot-preview[bot]');
+
+    expect(infoSpy).toHaveBeenCalledWith(
+      'Pull request cannot be merged cleanly. Current state: UNKNOWN.',
+    );
+  });
+
+  it('does not approve pull request which state is not open', async (): Promise<
```
Comment (merlinnot) [inline src/eventHandlers/checkSuite/index.spec.ts]:
```suggestion
  it('does not approve pull requests for which state is not open', async (): Promise<
```

File: src/eventHandlers/checkSuite/index.ts [hunk 10]
```diff
@@ -128,15 +127,18 @@ const tryMerge = async (
   }
 };
 
-export const checkSuiteHandle = async (octokit: GitHub): Promise<void> => {
+export const checkSuiteHandle = async (
+  octokit: GitHub,
+  GITHUB_LOGIN: string,
```
Comment (merlinnot) [inline src/eventHandlers/checkSuite/index.ts]: It's not a constant here, it's a parameter.

```suggestion
  gitHubLogin: string,
```

File: src/eventHandlers/pullRequest/index.ts [hunk 11]
```diff
@@ -55,7 +54,10 @@ const getPullRequestInformation = async (
   };
 };
 
-export const pullRequestHandle = async (octokit: GitHub): Promise<void> => {
+export const pullRequestHandle = async (
+  octokit: GitHub,
+  GITHUB_LOGIN: string,
```
Comment (merlinnot) [inline src/eventHandlers/pullRequest/index.ts]: Same here.

```suggestion
  gitHubLogin: string,
```

File: src/eventHandlers/push/index.spec.ts [hunk 12]
```diff
@@ -104,7 +104,135 @@ describe('push event handler', (): void => {
       })
       .reply(OK);
 
-    await pushHandle(octokit);
+    await pushHandle(octokit, 'dependabot-preview[bot]');
+  });
+
+  it('does not approve pull request that is not mergeable', async (): Promise<
```
Comment (merlinnot) [inline src/eventHandlers/push/index.spec.ts]:
```suggestion
  it('does not approve pull requests that are not mergeable', async (): Promise<
```

File: src/eventHandlers/push/index.spec.ts [hunk 13]
```diff
@@ -104,7 +104,135 @@ describe('push event handler', (): void => {
...
+      });
+
+    await pushHandle(octokit, 'dependabot-preview[bot]');
+
+    expect(infoSpy).toHaveBeenCalledWith(
+      'Pull request is not in a mergeable state: CONFLICTING.',
+    );
+  });
+
+  it('does not approve pull request that is already merged', async (): Promise<
```
Comment (merlinnot) [inline src/eventHandlers/push/index.spec.ts]:
```suggestion
  it('does not approve pull requests that are already merged', async (): Promise<
```

File: src/eventHandlers/push/index.ts [hunk 14]
```diff
@@ -118,9 +117,12 @@ const tryMerge = async (
   }
 };
 
-export const pushHandle = async (octokit: GitHub): Promise<void> => {
-  if (context.payload.pusher.name !== DEPENDABOT_GITHUB_LOGIN) {
-    logInfo('Pull request not created by Dependabot, skipping.');
+export const pushHandle = async (
+  octokit: GitHub,
+  GITHUB_LOGIN: string,
```
Comment (merlinnot) [inline src/eventHandlers/push/index.ts]: Same here.

```suggestion
  gitHubLogin: string,
```

File: src/index.ts [hunk 15]
```diff
@@ -1,4 +1,5 @@
 import { getInput, setFailed, warning } from '@actions/core';
+import { info } from '@actions/core/lib/core';
```
Comment (merlinnot) [inline src/index.ts]: Please don't import from internals of modules. We should consume only the advertised API.

File: src/index.ts [hunk 16]
```diff
@@ -10,15 +11,18 @@ import {
 const GITHUB_TOKEN = getInput('GITHUB_TOKEN');
 
 const octokit = new GitHub(GITHUB_TOKEN);
+const GITHUB_LOGIN = getInput('GITHUB_LOGIN');
```
Comment (merlinnot) [inline src/index.ts]: Should we move it before the definition of `GITHUB_TOKEN `? Seems to belong there, rather than here.

File: src/index.ts [hunk 17]
```diff
@@ -10,15 +11,18 @@ import {
...
 
 const main = async (): Promise<void> => {
+  info(`The bot name is ${GITHUB_LOGIN}`);
```
Comment (merlinnot) [inline src/index.ts]:
```suggestion
  info(`Automatic merges enabled for a bot with GitHub login ${GITHUB_LOGIN}.`);
```

Comment (viestat) [reply src/index.ts]: strictly speaking, this does not really tie to any bot, just a Github login.

File: src/index.ts (hunk 16 above)
Comment (viestat) [reply src/index.ts]: fixed with: https://github.com/ridedott/dependabot-auto-merge-action/pull/233/commits/8fa422481dda410cd7bc52b6c04ead5cbcab9c54

File: src/index.ts (hunk 15 above)
Comment (viestat) [reply src/index.ts]: fixed with: https://github.com/ridedott/dependabot-auto-merge-action/pull/233/commits/8fa422481dda410cd7bc52b6c04ead5cbcab9c54

File: src/eventHandlers/push/index.ts (hunk 14 above)
Comment (viestat) [reply src/eventHandlers/push/index.ts]: fixed with: https://github.com/ridedott/dependabot-auto-merge-action/pull/233/commits/8fa422481dda410cd7bc52b6c04ead5cbcab9c54

File: README.md (hunk 4 above)
Comment (viestat) [reply README.md]: fixed with https://github.com/ridedott/dependabot-auto-merge-action/pull/233/commits/3e01c991795106436edb97fcce1139496c4eb0c9

File: src/eventHandlers/checkSuite/index.spec.ts (hunk 6 above)
Comment (viestat) [reply src/eventHandlers/checkSuite/index.spec.ts]: fixed with https://github.com/ridedott/dependabot-auto-merge-action/pull/233/commits/8fa422481dda410cd7bc52b6c04ead5cbcab9c54

File: src/eventHandlers/checkSuite/index.ts (hunk 10 above)
Comment (viestat) [reply src/eventHandlers/checkSuite/index.ts]: fixed with https://github.com/ridedott/dependabot-auto-merge-action/pull/233/commits/8fa422481dda410cd7bc52b6c04ead5cbcab9c54

File: src/eventHandlers/pullRequest/index.ts (hunk 11 above)
Comment (viestat) [reply src/eventHandlers/pullRequest/index.ts]: fixed with https://github.com/ridedott/dependabot-auto-merge-action/pull/233/commits/8fa422481dda410cd7bc52b6c04ead5cbcab9c54

File: src/index.ts (hunk 17 above)
Comment (viestat) [reply src/index.ts]: https://github.com/ridedott/dependabot-auto-merge-action/pull/233/commits/6ccdba337490234b2a3d3a9d346c82c38a88185c WDYT?

File: src/eventHandlers/checkSuite/index.spec.ts [hunk 18]
```diff
@@ -114,7 +112,207 @@ describe('check Suite event handler', (): void => {
       })
       .reply(OK);
 
-    await checkSuiteHandle(octokit);
+    await checkSuiteHandle(octokit, 'dependabot-preview[bot]');
+  });
+
+  it('does not approve pull request that is not mergeable', async (): Promise<
```
Comment (merlinnot) [inline src/eventHandlers/checkSuite/index.spec.ts]:
```suggestion
  it('does not approve pull requests that are not mergeable', async (): Promise<
```

**Pull request merged**